### PR TITLE
Activate/fix rpy2 tests

### LINKED ISSUE #35347: `rpy2`: Document that it is a "semistandard" package
... in the `SPKG.rst` of `r` and `rpy2`.

Also, for dummy packages such as `r`, do not show the message "If the system package is installed, ./configure will
check whether it can be used."

see https://groups.google.com/g/sage-devel/c/0Rl6YbnPNmM

Also use the `feature` argument in the `lazy_import` of `rpy2` in `sage.interfaces.r`, so that system package info will be displayed when rpy2 is not installed.

### PR DESCRIPTION
**Pull request by mkoeppe**
### :books: Description

As a follow-up on:
- #35237 

we make `rpy2` a dynamically detected feature, enabling the automatic run of doctests tagged `# optional - rpy2`; we use a file-level doctest tag for the `r.py` file.

We also update the spkg documentation for #35347

Resolves #35347

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### :hourglass: Dependencies

- Depends on #35237

### CODE REVIEW

File: src/sage/interfaces/r.py [hunk 1]
```diff
@@ -1164,8 +1158,8 @@ def completions(self, s):
 
         EXAMPLES::
 
-            sage: dummy = r._tab_completion(use_disk_cache=False)    #clean doctest  # optional - rpy2
-            sage: 'testInheritedMethods' in r.completions('tes')  # optional - rpy2
+            sage: dummy = r._tab_completion(use_disk_cache=False)    #clean doctest
+            sage: 'testInheritedMethods' in r.completions('tes')
```
Comment (kwankyu) [inline src/sage/interfaces/r.py:1162]:
```
# clean doctest
```

File: src/sage/interfaces/r.py [hunk 2]
```diff
@@ -1313,13 +1306,12 @@ def plot(self, *args, **kwds):
         would need to use the following since R lattice graphics do
         not automatically print away from the command line::
 
-            sage: # optional - rpy2
-            sage: filename = tmp_filename() + '.png' # Not needed in notebook, used for doctesting
-            sage: r.png(filename='"%s"'%filename)       # optional - rgraphics
+            sage: filename = tmp_filename() + '.png'  # Not needed in notebook, used for doctesting
+            sage: r.png(filename='"%s"'%filename)               # optional - rgraphics
```
Comment (kwankyu) [inline src/sage/interfaces/r.py:1310]:
```
# not needed in notebook, used for doctesting
```

File: src/sage/interfaces/r.py [hunk 3]
```diff
@@ -1274,7 +1268,7 @@ def plot(self, *args, **kwds):
         persistent in your own code::
 
             sage: from tempfile import TemporaryDirectory
-            sage: with TemporaryDirectory() as d: # optional - rpy2, rgraphics
+            sage: with TemporaryDirectory() as d:               # optional - rgraphics
             ....:     _ = r.setwd(d)
```
Comment (kwankyu) [inline src/sage/interfaces/r.py:1272]: Where is `rgraphics` feature defined?

Comment (mkoeppe) [reply src/sage/interfaces/r.py:1272]: It isn't anywhere

Comment (kwankyu) [reply src/sage/interfaces/r.py:1272]: Then does `# optional - rgraphics` work?

Comment (mkoeppe) [reply src/sage/interfaces/r.py:1272]: Well, people who know about it can use `sage -t --optional=sage,rpy2,rgraphics`, I suppose.

Comment (kwankyu) [reply src/sage/interfaces/r.py:1272]: Perhaps I forgot how the `optional` mechanism is supposed to work...

Comment (kwankyu) [reply src/sage/interfaces/r.py:1272]: OK. What you said seems the best we can suppose.

File: src/sage/interfaces/r.py (hunk 2 above)
Comment (kwankyu) [reply src/sage/interfaces/r.py:1310]: Perhaps not in the scope, as there are many more such comments.

**Comment by kwankyu** [comment]
Otherwise lgtm.

**Comment by kwankyu** [comment]
There is a genuine failure:
```
sage -t --random-seed=30511447838430634235757231865836227899 src/sage/interfaces/r.py
**********************************************************************
File "src/sage/interfaces/r.py", line 1312, in sage.interfaces.r.R.plot
Failed example:
    r.library("lattice")
Exception raised:
    Traceback (most recent call last):
      File "/Users/runner/work/sage/sage/src/sage/doctest/forker.py", line 714, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/Users/runner/work/sage/sage/src/sage/doctest/forker.py", line 1146, in compile_and_execute
        exec(compiled, globs)
      File "<doctest sage.interfaces.r.R.plot[5]>", line 1, in <module>
        r.library("lattice")
      File "/Users/runner/work/sage/sage/src/sage/interfaces/r.py", line 860, in library
        raise ImportError("R library {} not installed".format(library_name))
    ImportError: R library lattice not installed
```

**Comment by mkoeppe** [comment]
Is that from the conda test? 

I added the package name here, but that will only take effect after updating the conda-lock files (as done and to be redone in #37998)

**Comment by kwankyu** [comment]
OK. 

Thanks.

**Review by kwankyu — APPROVED** [review]

**Comment by mkoeppe** [comment]
Thanks!

**Pull request merged**